.github/workflows/release.yaml

```diff
@@ -0,0 +1,912 @@
+      ARCH: 64
+      TARBALL_EXT: tar.xz
+      DISTRO: na
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: bindists-mac-x86_64
+          path: ./out
+
+      - name: Run test (mac)
+        run: |
+          brew install coreutils tree
+          bash .github/scripts/test.sh
+
+  test-mac-aarch64:
+    name: Test binary (Mac aarch64)
+    runs-on: [self-hosted, macOS, ARM64]
+    needs: ["bindist-mac-aarch64"]
+    env:
+      MACOSX_DEPLOYMENT_TARGET: 10.13
+      ARTIFACT: "aarch64-apple-darwin"
+      ARCH: ARM64
+      TARBALL_EXT: tar.xz
+      DISTRO: n
+      HOMEBREW_CHANGE_ARCH_TO_ARM: 1
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: bindists-mac-aarch64
+          path: ./out
+
+      - name: Run test (mac)
+        run: |
+          bash .github/scripts/brew.sh git coreutils llvm@11 autoconf automake tree
+          export PATH="$HOME/.brew/bin:$HOME/.brew/sbin:$HOME/.brew/opt/llvm@11/bin:$PATH"
+          export CC="$HOME/.brew/opt/llvm@11/bin/clang"
+          export CXX="$HOME/.brew/opt/llvm@11/bin/clang++"
+          export LD=ld
+          export AR="$HOME/.brew/opt/llvm@11/bin/llvm-ar"
+          export RANLIB="$HOME/.brew/opt/llvm@11/bin/llvm-ranlib"
+          bash .github/scripts/test.sh
+
+  test-win:
+    name: Test binary (Win)
+    runs-on: windows-latest
+    needs: ["bindist-win"]
+    env:
+      ARTIFACT: "x86_64-mingw64"
+      ARCH: 64
+      TARBALL_EXT: zip
+      DISTRO: na
+    strategy:
+      fail-fast: false
+    steps:
+      - name: install windows deps
+        shell: pwsh
+        run: |
+          C:\msys64\usr\bin\bash -lc "pacman --disable-download-timeout --noconfirm -Syuu"
+          C:\msys64\usr\bin\bash -lc "pacman --disable-download-timeout --noconfirm -Syuu"
+          C:\msys64\usr\bin\bash -lc "pacman --disable-download-timeout --noconfirm -S make mingw-w64-x86_64-clang curl autoconf mingw-w64-x86_64-pkgconf ca-certificates base-devel gettext autoconf make libtool automake python p7zip patch unzip zip git"
+          taskkill /F /FI "MODULES eq msys-2.0.dll"
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: bindists-win
+          path: ./out
+
+      - name: Run test (windows)
+        run: |
+          $env:CHERE_INVOKING = 1
+          $env:MSYS2_PATH_TYPE = "inherit"
+          C:\msys64\usr\bin\bash -lc "bash .github/scripts/test.sh"
+        shell: pwsh
+
+  release:
+    name: release
+    needs: ["test-linux", "test-mac-x86_64", "test-mac-aarch64", "test-win", "test-arm"]
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/')
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          path: ./out
+          name: bindists-x86_64-linux-deb9
```

Comment:
The literal listing of all the artifact names makes me sad

Reply:
Could we not use the download-all ability of `download-artifact`? Since you've proactively deleted all the non-bindist artifacts by this point.

Reply:
No, that will make the runners run out of disk space.